bot/cogs/weather_cog.py

```diff
@@ -0,0 +1,353 @@
+                    if i > 1:
+                        page += f'{date_str}: {round(day_temp,1)}°F\n'
+                        page += f'{date_str} Night: {round(night_temp,1)}°F\n'
+                    elif i == 1:
+                        page += f'Tomorrow: {round(day_temp,1)}°F\n'
+                        page += f'Tomorrow Night: {round(night_temp,1)}°F\n'
+                    else:
+                        page += f'Today: {round(day_temp,1)}°F\n'
+                        page += f'Tonight: {round(night_temp,1)}°F\n'
+                else:
+                    page += f'Time: {time_str}\n'
+                    page += f'Temperature: {round(temp,1)}°F\n'
+                
+                page += f'Condition: {desc}\n\n'
+                page += f'Chance of Precipitation: {round(precp*100)}%\n'
+                page += f'Humidity: {round(hum)}%\n'
+                page += f'Wind: {round(wind,1)} mph ({wind_dir})\n\n'
+                pages.append(page)
+        
+        return pages, num_hr, num_day
+
+    async def weatherCode(self, ctx, loc, is_cond, is_hr, is_day):
+        # Remove any characters not in ranges a-z, A-Z, or 0-9
+        # Exceptions: & _ - , and <space>
+        # per the ASCII Table https://www.asciitable.com
+        loc = re.sub("[^a-zA-Z0-9 ,&_-]+", "", loc)
+
+        # Geocoding URL
+        url_GeoAPI = f'{url_Geo}{loc}'
+
+        self.geocode_api_key = BotSecrets.get_instance().geocode_key
+        self.weather_api_key = BotSecrets.get_instance().weather_key
+
+        geo_queryparams = {
+            'auth' : self.geocode_api_key,
+            'json' : '1',
+            }
+
+        # Message to Display while APIs are called
+        wait_msg = await ctx.send('Converting location')
+
+        # Try Except for catching errors that could give away either API key
+        try:
+            async with aiohttp.request("GET", url_GeoAPI, params=geo_queryparams) as response:
+                if (response.status == 200):
+                    res_geo_json = await response.json()
+                    city = res_geo_json.get('standard',{}).get('city',{})
+                    lon = res_geo_json.get('longt',{})
+                    lat = res_geo_json.get('latt',{})
+
+                    queryparams = {
+                        'lat' : lat,
+                        'lon' : lon,
+                        'appid' : self.weather_api_key,
+                        'units' : 'imperial',
+                        'lang' : 'en'
+                        }
+
+                    weatherPages = []
+                    await wait_msg.edit(content='Checking the weather')
+                    
+                    async with aiohttp.request("GET", url, params=queryparams) as response:
+                        if (response.status == 200):
+                            res_json = await response.json()
+                            weatherPages, num_hr, num_day = self.getPageData(lat, lon, res_json, city, \
```

Comment:
instead of using \ we should break on the , to create a nicely formatted call, see the bottom of clem_bot.py for an example of what im talking about